examples/main/main.cpp

```diff
@@ -218,24 +146,23 @@ int main(int argc, char ** argv) {
         return 1;
     }
 
-    params.n_keep    = std::min(params.n_keep,    (int) embd_inp.size());
+    // number of tokens to keep when resetting context
+    if (params.n_keep < 0 || params.n_keep > (int)embd_inp.size() || params.instruct) {
+        params.n_keep = (int)embd_inp.size();
+    }
```

Comment:
Update `gpt_print_usage` to explain that `--keep -1` will use the entire prompt automatically (and also show what is the default, which is currently missing)

Reply:
:+1: